File: services/geoapi/geoapi/Storages/ITerrainObjectStore.cs
using System.Collections.Generic;
using geoapi.Models;

namespace geoapi.Storages
{
    public interface ITerrainObjectStore
    {
        TerrainObject GetTerrainObject(string objectId);
        void UploadTerrainObject( TerrainObject terrainObject);
        IEnumerable<TerrainObject> GetTerrainObjects(string ownerIndex, int skip, int take);
    }
}

File: services/geoapi/geoapi/Storages/ITerrainObjectStore.vb
Imports System.Collections.Generic
Imports geoapi.Models

Namespace Storages
    Public Interface ITerrainObjectStore
        Function GetTerrainObject(ByVal key As String) As TerrainObject
        Sub UploadTerrainObject(ByVal terrainObject As TerrainObject)
        Function GetTerrainObjects(ByVal agentToken As String, ByVal skip As Integer, ByVal take As Integer) As IEnumerable(Of TerrainObject)
    End Interface
End Namespace
